subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -1367,3 +1367,46 @@ Gradle 5.3 introduced the interface `TransformAction` for implementing artifact
 Using the registration method link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(org.gradle.api.Action)[DependencyHandler.registerTransform(Action)] for `ArtifactTransform` has been deprecated.
 Migrate your artifact transform to use `TransformAction` and use link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(java.lang.Class,%20org.gradle.api.Action)[DependencyHandler.registerTransform(Class, Action)] instead.
 See the <<artifact_transforms#sec:abm_artifact_transforms,user manual>> for more information on implementing `TransformAction`.
+
+[[gmm_modification_after_publication_populated]]
+==== Modifying GMM after a publication has been populated
+
+Altering the link:publishing_gradle_module_metadata.html[Gradle Module Metadata] (e.g. changing a component configuration variants) after a Maven or Ivy publication have been populated from their components, is now deprecated and scheduled to be disallowed in Gradle 9.0.
+
+Eager "realization" of the publication can happen if the following methods are called:
+
+====== Maven
+
+* link:{javadocPath}/org/gradle/api/publish/maven/MavenPublication.html#getArtifacts--[MavenPublication.getArtifacts()]
+
+====== Ivy
+
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#getArtifacts--[IvyPublication.getArtifacts()]
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#getConfigurations--[IvyPublication.getConfigurations()]
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#configurations(Action)--[IvyPublication.configurations(Action)]
+
+Before, it was possible to have this code block without any warnings, but it created inconsistencies between the published artifacts:
```

Comment:
```suggestion
Previously, this code block did not generate warnings, but it created inconsistencies between published artifacts:
```
Suggestion: reword to reduce wordiness. Consider elaborating on "inconsistencies."